lib/util.ex

```diff
@@ -67,27 +77,33 @@ defmodule Util do
       Authorization: "Bearer " <> auth
     ]
 
-    {:ok, resp} = HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers)
-
-    resp.body
-    |> JSON.decode!()
-    |> Map.get("data")
-    |> Map.get("tournaments")
-    |> Map.get("nodes")
-    # Finesse data into our own non-start.gg format.
-    |> Enum.map(fn sgg_data ->
-      %{
-        "external_id" => "smashgg-" <> Integer.to_string(sgg_data["id"]),
-        "name" => sgg_data["name"],
-        "location" => %{
-          "lat" => sgg_data["lat"],
-          "lng" => sgg_data["lng"]
-        },
-        "start_time" => sgg_data["startAt"],
-        "end_time" => sgg_data["endAt"],
-        "updated_at" => sgg_data["updatedAt"],
-        "url" => "https://start.gg/" <> (sgg_data["slug"])
-      }
-    end)
+    case HTTPoison.post("https://api.start.gg/gql/alpha", JSON.encode!(body), headers) do
+      {:ok, resp} ->
+        try do
+          resp.body
+          |> JSON.decode!
+          |> Map.get("data")
+          |> Map.get("tournaments")
+          |> Map.get("nodes")
+          # Finesse data into our own non-start.gg format.
+          |> Enum.map(fn sgg_data ->
+            %{
+              "external_id" => "smashgg-" <> Integer.to_string(sgg_data["id"]),
+              "name" => sgg_data["name"],
+              "location" => %{
+                "lat" => sgg_data["lat"],
+                "lng" => sgg_data["lng"]
+              },
+              "start_time" => sgg_data["startAt"],
+              "end_time" => sgg_data["endAt"],
+              "updated_at" => sgg_data["updatedAt"],
+              "url" => "https://start.gg/" <> (sgg_data["slug"])
+            }
+            end )
+          rescue
+            _e -> raise "Ran into an issue while parsing the response body in get_tourney_page function."
```

Comment:
Unless I'm misunderstanding something, this is actually hiding whatever error caused us to get here. We want more info being printed, not less!

Reply:
Yeah you're totally right, a BadMapError was occurring and I thought I'd be able to provide more context, not the way to go lol.

Reply:
I mean adding context is usually a good thing, the real problem (other than `try...rescue`) is that this would have suppressed the original error. It's fine to add info about what we were doing when we noticed the error, but the error itself might contain valuable debugging info that we wouldn't have wanted to discard.